v2.2/query-behavior-troubleshooting.md

```diff
@@ -1,31 +1,94 @@
 
-- [Turn on SQL logging](#sql-logging).
+If you need help interpreting the output of the `EXPLAIN ANALYZE` statement, contact us.
 
-If you're still unable to determine why the query executes slowly, please [file an issue](file-an-issue.html).
+## Query is sometimes slow
+
+If the query performance is irregular:
+
+1.  Run [`SHOW TRACE`](https://www.cockroachlabs.com/docs/stable/show-trace.html) for the query twice: once when the query is performing as expected and once when the query is slow.
+
+2.  Contact us to analyze the outputs of the `SHOW TRACE` command.
+
+## Cancelling running queries
+
+See [Cancel query](https://www.cockroachlabs.com/docs/v2.1/manage-long-running-queries.html#cancel-long-running-queries)
+
+## Low throughput
+
+Throughput is affected by the Disk I/O, CPU usage, and Network latency. Use the Admin UI to check the following metrics:
+
+Disk I/O: [Disk IOPS in progress](https://www.cockroachlabs.com/docs/v2.1/admin-ui-hardware-dashboard.html#disk-iops-in-progress)
+
+CPU usage: [CPU percent](https://www.cockroachlabs.com/docs/v2.1/admin-ui-hardware-dashboard.html#cpu-percent)
+
+Network latency: [Service latency SQL 99th percentile](https://www.cockroachlabs.com/docs/v2.1/admin-ui-overview-dashboard.html#service-latency-sql-99th-percentile)
+
+## Single hot node: One node has much higher resource usage than other nodes
+
+To determine if you have a hot node in your cluster, [access the Admin UI](admin-ui-access-and-navigate.html#access-the-admin-ui), click **Metrics** on the left, and navigate to the following graphs. Hover over each of the following graphs to see the per-node values of the metrics. If one of the nodes has a higher value, you have a hot node in your cluster.
+
+-   Replication dashboard > Average queries per store graph.
+
+-   Overview Dashboard > Service Latency graph
+
+-   Hardware Dashboard > CPU percent graph
+
+-   SQL Dashboard > SQL Connections graph
+
+-   Hardware Dashboard > Disk IOPS in Progress graph
+
+**Solution:**
+
+-   If you have a small table that fits into one range, then only one of the nodes will be used. This is expected behavior.
+
+-   If the **SQL Connections** graph shows that one node has a higher number of SQL connections and other nodes have zero connections, check if your app is set to talk to only one node.
+
+-   Check load balancer settings.
+
+-   Check for [transaction contention](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#understanding-and-avoiding-transaction-contention).
+
+-   Use `UUID` instead of `SERIAL`. See [UUID Best practices](https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#unique-id-best-practices) for more information.
```

Comment:
Without more context, this seems like it would be really challenging for users to understand.

Reply:
The linked section provides detailed info.